### PR TITLE
DS-704, DS-708 | @ericbakenhus | Nested filters and online options facet

### PR DESCRIPTION
**Pull request by ericbakenhus**
# READY FOR REVIEW

# Summary
- Refactor useFacets
- Adds nested filter
- Adds online options filter

# Review By (Date)
- 🤷 

# Criticality
- Low

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch or use the deploy preview
2. Navigate to [/events/discovery](https://deploy-preview-875--stanford-alumni.netlify.app/events/discovery/)
3. Verify the Online options filter is working
4. Verify all the accordions and new sub accordion is working as expected

# Associated Issues and/or People
- DS-708
- DS-704

### CODE REVIEW

**Comment by sherakama** [comment]
Missing the reset filters link in the nested option

Screenshot 2024-07-05 at 15 27 33

**Review by sherakama — APPROVED** [review]
Throw in a reset filter link on the sub-facet group and this is GTG.

**Comment by sherakama** [comment]
I am making the assumption that the mobile version of the sub-facet is a WIP.

Screenshot 2024-07-05 at 15 52 53

**Comment by ericbakenhus** [comment]
> Missing the reset filters link in the nested option

Would be helpful if I actually implemented the reset button and didn't just lie to myself with through jsdocs. 😆 

Should be in there now.

> I am making the assumption that the mobile version of the sub-facet is a WIP.

Yes, I'm handling the remaining mobile facet work in DS-716.

File: src/components/events-discovery/components/DateFilter/DateFilter.jsx [hunk 1]
```diff
@@ -285,6 +286,8 @@ export const DateFilter = ({ expanded, toggleFacet = () => null }) => {
                     maxDate={dateEnd || midnight.plus({ days: 90 })}
                     onChange={handleDateStartChange}
                     timezone="system"
+                    views={['day']}
+                    format="MM/dd/yyyy"
```
Comment (ericbakenhus) [inline src/components/events-discovery/components/DateFilter/DateFilter.jsx:290]: From DS-706.

File: src/components/events-discovery/components/Facets/FacetCtx.jsx
Comment (ericbakenhus) [inline src/components/events-discovery/components/Facets/FacetCtx.jsx:1]: Moving the previous face hook state into some shared context.

File: src/components/events-discovery/components/FilterAccordion/SubfilterAccordion.jsx
Comment (ericbakenhus) [inline src/components/events-discovery/components/FilterAccordion/SubfilterAccordion.jsx:1]: Completed the old for the sub-filter accordion.

File: src/components/events-discovery/components/LocationFilter/LocationFilter.jsx
Comment (ericbakenhus) [inline src/components/events-discovery/components/LocationFilter/LocationFilter.jsx:1]: New location filter.

File: src/components/events-discovery/components/LocationFilter/OnlineOptionsSubfilter.jsx
Comment (ericbakenhus) [inline src/components/events-discovery/components/LocationFilter/OnlineOptionsSubfilter.jsx:1]: Moved online options into this location sub-filter.

File: src/components/events-discovery/eventsDiscovery.jsx [hunk 2]
```diff
@@ -32,30 +33,32 @@ const EventDiscoveryContent = () => {
...
-    <div className="su-cc lg:su-flex lg:su-gap-x-40 su-mx-12">
-      <div className="su-hidden lg:su-block su-flex-none su-w-300">
-        <DesktopFilter />
-      </div>
-      <div className="su-flex su-flex-col su-gap-y-20 su-w-full">
-        <StatusHeader hitsPerPage={hitsPerPage} />
-        <div className="su-flex su-flex-none su-flex-row">
-          <ChipsComponent />
-          <MobileFilter />
+    <FacetProvider>
```
Comment (ericbakenhus) [inline src/components/events-discovery/eventsDiscovery.jsx:36]: Only change here is wrapping it with the facet provider component.

File: src/components/events-discovery/components/DateFilter/DateFilter.jsx (hunk 1 above)
Comment (sherakama) [reply src/components/events-discovery/components/DateFilter/DateFilter.jsx:290]: Ty

**Pull request merged**